pkg/estimator/client/interface.go

```diff
@@ -53,3 +54,16 @@ func GetReplicaEstimators() map[string]ReplicaEstimator {
 func GetUnschedulableReplicaEstimators() map[string]UnschedulableReplicaEstimator {
 	return unschedulableReplicaEstimators
 }
+
+// EstimatorPriority
+// 1. Higher-priority estimator can directly override the result given by lower-priority estimator
+// 2. Lower replicas estimated will be chosen if the priority of estimator are same
+// 3. Lower-priority estimator would be ignored if the result was already given by higher-priority estimator
+type EstimatorPriority int
+
+const (
+	Undefined EstimatorPriority = iota
```

Comment:
mainly because:

https://github.com/karmada-io/karmada/blob/1cd8b0b1be4b29f0a0c7b8a02c750803776d8537/pkg/scheduler/core/util.go#L68-L70

this slice has default value 0, I don't want it equal to any valid priority.

may be I can remove `Undefined` priority and  change above line as:

```go
for i := range availableResultPriority { 
 	availableResultPriority[i] = -1
 } 
```

do you perfer I deleting the `Undefined` priority?

Reply:
deleting the `Undefined` priority is ok. BTW, could you consider designing priorities to be more scalable? if I want to add a new estimator which priority is between `SchedulerEstimator` and `GeneralEstimator`, how should I expand it?

Reply:
> if I want to add a new estimator which priority is between SchedulerEstimator and GeneralEstimator, how should I expand it?

You are absolutely right, thank you very much for your advice~